auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/AuthorizationClient.java

```diff
@@ -48,6 +48,24 @@ public void checkRight(String requestId, String userCrn, String right, String re
         );
     }
 
+    public boolean hasRight(String requestId, String userCrn, String right, String resource) {
+        checkNotNull(requestId);
+        checkNotNull(userCrn);
+        checkNotNull(right);
+
+        AuthorizationProto.HasRightsRequest.Builder requestBuilder = AuthorizationProto.HasRightsRequest.newBuilder()
+                .setActorCrn(userCrn);
+
+        AuthorizationProto.RightCheck.Builder rightCheckBuilder = AuthorizationProto.RightCheck.newBuilder().setRight(right);
+        if (!StringUtils.isEmpty(resource)) {
+            rightCheckBuilder.setResource(resource);
+        }
+        requestBuilder.addCheck(rightCheckBuilder.build());
+
+        AuthorizationProto.HasRightsResponse response = newStub(requestId).hasRights(requestBuilder.build());
+        return response.getResultList().get(0);
```

Comment:
So while this is fine it means that we are doing some extra work. Because I believe you are also bringing the actor's rights with the call to GetRights somewhere (you need it for the group membership and workload-administration-group membership information). Moreover, there is a potential for discrepancy between the two calls - the actor rights may have change in between. So I think it is better to use the actor rights you get from the GetRights response to check for the rights the actor has. Let's talk about this offline I am not sure how much of the scrutinizer was ported to CB code base.